### ISSUE TITLE
BUG: Itemized Menus

### ISSUE THREAD

**Issue by giom-mallet** [pacollins/hugo-future-imperfect-slim#119]
# Bug Report
## Describe the bug
Hi, this is probably more likely a question, but  I am confused by one part of the behaviour of the Template.
I am trying a to create a new sub-folder in my content, which I intend to be a short series of pages not directly related to other posts.
I created the new folder in content (let's call it /itemized, with two new .md (item1.md and item2.md) files inside.
I created the Menu item in config.toml, and manage to get it in the bar.
Accessing the new pages through /itemized/item1 works fine.
But the page at /itemized instead of showing me my two new items only shows me the other posts ( currently the ones from the Example Site. 
If I add 
` type:`` "post"

 in the front matter, now my new items show up in the list as well, but it is not what I would expect. Adding 
` type: "itemized"

doesn't seem to work either. 

## To Reproduce

Create a new content subfolder with more than one item. 
Build the website, of run with the test server. 

## Expected behavior
Issue might come from my lack of understanding of Hugo, I only made a website two years ago using the Ananke theme, and don't recall running into this problem with lists. 

Ideally, I think an ExampleWebsite with another Menu on top of "Blog" might be super helpful !

## Environment

 - **Hugo Version:** 0.65.3
 - **Browser and Version:** Firefox 0.73
 - **Platform:** Desktop
 - **Link to Project:**

**Comment by pacollins** [comment]
I am sure this bug was introduced in #71. I am looking at it, but can't seem to find the right solution. I will have to look a little.

Good catch.

**Comment by pacollins** [comment]
I have found the issue - we prescribe too much specificity in the theme. I will make the changes, but it might be two weeks as I am very busy of the next period of time.

**Comment by giom-mallet** [comment]
Good news, thank you

I managed to work around this by creating a menu to a category page, but actually even in these pages, only the articles with the type "post will show.
I am mostly working on that during week ends, but don't hesitate if you want me to test your latest work.

Note : I am in China, I can probably help you to get the 'Wechat Integration" button work better. If you want. 

> 
> 
> I have found the issue - we prescribe too much specificity in the theme. I will make the changes, but it might be two weeks as I am very busy of the next period of time.

**Comment by matt9j** [comment]
Hi @giom-mallet , I just opened a PR that addresses the "too much specificity" issues @pacollins mentioned above. If you apply #122 and then add your `itemized` content type to the params.mainSections parameter in your site's config.coml I think things should work correctly for non-post content types.

**Comment by giom-mallet** [comment]
> 
> 
> Hi @giom-mallet , I just opened a PR that addresses the "too much specificity" issues @pacollins mentioned above. If you apply #122 and then add your `itemized` content type to the params.mainSections parameter in your site's config.coml I think things should work correctly for non-post content types.

Hi @matt9j 

Thank you, I switched the theme to your branch, but I am afraid the result is still not what I would expect : 
Now as I put "itemized" in params.mainSections and in the type of my new articles,
 When I navigate to /blog and see both my types : "post" and "itemized" 
  Same when I navigate to /itemized I see all articles ( type: "blog" and type:"itemized" )

What I am not sure to understand, is why not simply using the folder in which the pages are stored to generate those list pages ? ( actually, "itemized" is not content I want to show in my home page)

**Comment by matt9j** [comment]
Ah, I think I understand now. The theme right now does not define a separate homepage template, and instead modifies the "list" template to include content from all of the mainSections. I made a test  branch that defines the homepage separately (with all posts from all mainSections), and leaves the list template to only show pages in a particular section. @giom-mallet, can check out matt9j:test-sections and see if that does what you need?

**Comment by giom-mallet** [comment]
Hi @matt9j, 
Thanks for your message. 

I switched to the matt9j:test-sections branch and I changed my articles to type 
"itemized"

Now I've got the following issues : 
  going to /itemized page is showing me unformatted content : 

> http://localhost:1313/itemized/item1/  Title1 on Site Hugo -- gohugo.io fr Sun, 01 Mar 2020 11:18:56 +0800 http://localhost:1313itemized/item2/ Recent content in Title2 on Site 2020 Hugo -- gohugo.io fr Sun, 01 Mar 2020 11:18:55 +0800 http://localhost:1313itemized/item3/  Recent content in Title3 on Site Hugo -- gohugo.io fr Sun, 01 Mar 2020 11:18:54 +0800 http://localhost:1313itemized/item4/  Recent content in Title1 on Site Hugo -- gohugo.io fr Sun, 01 Mar 2020 11:15:53 +0800 http://localhost:1313itemized/item5/ Recent content in Title5 on Site Hugo -- gohugo.io fr Sun, 01 Mar 2020 11:15:52 +0800 2020 11:15:51 +0800 

It seems to have also impacted the pages themselves : http://localhost:1313/itemized/item1/ have an image defined, but it doesn't display anymore.

Hope this helps

**Comment by matt9j** [comment]
Hmm, interesting! I got a second to look at this this morning, and found a couple things.

1) There appears to be a hugo template lookup inconsistency where if a featured image is defined, *but the path is wrong* that it will fail to find and render the content-list layout. This results in the unformatted text you see in the localhost:1313/itemized list view. What's really odd, is that if you fix the picture link *the debug server detects the change and refreshes, but this does not fix the rendering issue*. I had to fully stop and start the debug server after fixing the path or removing the featured image parameter, and only then would the template render correctly.

2) The reason the featured picture did not appear in the single post view is unrelated, is that the single view for default content types is much simpler than "post" and does not have a featured image (nor social media sharing, nor comments...). I made a commit that makes the default type as fully featured as post (except for a post specific header with reading time) and pushed it to the same test branch (matt9j:test-sections). Take a look and see if that addresses your issues! If so, the maintainers will have to decide if that's how they want to organize things in the theme.
